ms2/test/src/org/labkey/test/tests/ms2/MS2Test.java

```diff
@@ -1224,4 +1222,33 @@ public Set<String> getFilters()
         }
         return result;
     }
+
+    /**
+     * Issue 52245 - container filter not getting propagated to run group lookup
+     */
+    private void validateContainerFilter()
+    {
+        // create a run group at the project level
+        goToProjectHome(getProjectName());
+        clickAndWait(Locator.linkWithText("Run Groups"));
+        clickButton("Create Run Group");
+        setFormElement(Locator.name("name"), RUN_GROUP3_NAME);
+        clickButton("Submit");
+
+        navigateToFolder(FOLDER_NAME);
+        DataRegionTable runsTable = new DataRegionTable(REGION_NAME_SEARCH_RUNS, this);
+        runsTable.checkAllOnPage();
+        runsTable.clickHeaderMenu("Add to run group", RUN_GROUP3_NAME);
+
+        goToProjectHome(getProjectName());
+        goToSchemaBrowser();
+        DataRegionTable dataTable = viewQueryData("ms2", "XTandemPeptides");
+        dataTable.setContainerFilter(DataRegionTable.ContainerFilterType.CURRENT_AND_SUBFOLDERS);
+
+        _customizeViewsHelper.openCustomizeViewPanel();
+        _customizeViewsHelper.addColumn(new String[]{"Fraction", "Run", "ExperimentRunLSID", "RunGroups"});
+        _customizeViewsHelper.applyCustomView();
+        dataTable.setFilter("Fraction/Run/ExperimentRunLSID/RunGroups", "Is Blank");
+        assertEquals("All rows should have a value for the run group", 0, dataTable.getDataRowCount());

```

Comment:
This should check for a non-zero row count before setting the filter. Just as a sanity check that filtering the view actually changes what's shown.
```suggestion
        _customizeViewsHelper.openCustomizeViewPanel();
        _customizeViewsHelper.addColumn(new String[]{"Fraction", "Run", "ExperimentRunLSID", "RunGroups"});
        _customizeViewsHelper.applyCustomView();
        assertNotEquals("All rows should have a value for the run group", 0, dataTable.getDataRowCount());
        dataTable.setFilter("Fraction/Run/ExperimentRunLSID/RunGroups", "Is Blank");
        assertEquals("All rows should have a value for the run group", 0, dataTable.getDataRowCount());

```

Reply:
good point, thanks